kyaml/fn/framework/framework.go

```diff
@@ -168,5 +319,130 @@ func (rl *ResourceList) Filter(api kio.Filter) error {
 		}
 		return errors.Wrap(err)
 	}
+	rl.Items = yaml.NewKubeObjectsFromRNodes(items)
+	return nil
+}
+
+// SortItems sorts the ResourceList.items by apiVersion, kind, namespace and name.
+func (rl *ResourceList) SortItems() {
+	sort.Sort(kubeObjects(rl.Items))
+}
+
+type kubeObjects []*yaml.KubeObject
+
+func (o kubeObjects) Len() int      { return len(o) }
+func (o kubeObjects) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o kubeObjects) Less(i, j int) bool {
+	idi := o[i].ResourceIdentifier()
+	idj := o[j].ResourceIdentifier()
+	idStrI := fmt.Sprintf("%s %s %s %s", idi.GetAPIVersion(), idi.GetKind(), idi.GetNamespace(), idi.GetName())
+	idStrJ := fmt.Sprintf("%s %s %s %s", idj.GetAPIVersion(), idj.GetKind(), idj.GetNamespace(), idj.GetName())
+	return idStrI < idStrJ
+}
+
+// AsMain reads the resourceList in yaml format from stdin, evaluates the
+// function and write the updated resourceList in yaml to stdout. Errors if any
+// will be printed to stderr.
+func AsMain(p ResourceListProcessor) error {
```

Comment:
This strikes me as a less powerful version of the `command` subpackage, which is intended to make it easy to generate entrypoints: https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml@v0.13.1/fn/framework/command. If we need this as well (please comment as to why), I think it should move to that subpackage, which is where we've isolated assumptions that the IO is from the command line.